tools/ci_build/build.py

```diff
@@ -904,7 +911,7 @@ def use_dev_mode(args):
         return False
     if args.use_armnn:
         return False
-    if args.ios and is_macOS():
+    if args.ios and is_macOS() and args.visionos:
```

Comment:
Should this be `(ios or visionos) and macos`?

Reply:
will update this